### PR TITLE
container - add hugepages config

### PR DESCRIPTION
**Pull request by DrFaust92**
Closes https://github.com/hashicorp/terraform-provider-google/issues/16497

tested locally:

```
--- PASS: TestAccContainerNodePool_withHugepageConfig (817.36s)
```

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `hugepages_config` in `linux_node_config` in `google_container_node_pool` resource
```

### CODE REVIEW

**Review by wyardley — COMMENTED** [review]
Not from Google, but just pointing out one minor thing in the test.

Only other thought: if this is valid in `google_container_cluster.node_config` as well (which I think it is), that should probably have a separate update test; also, I noticed with #11272 at least, that updating the default nodepool when used in that context isn't always setup properly.

May be slightly different for this one, but I had to add some changes in [here](https://github.com/GoogleCloudPlatform/magic-modules/blob/b7538e9de350e770ddedc738e23094348746681b/mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb#L3800-L3885) for that to work (assuming either the nodepool or the cluster needs to get updated there).

**Comment by melinath** [comment]
Apologies for the delayed review - reassigning since I'll be OOO tomorrow.

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb [hunk 1]
```diff
@@ -656,6 +656,40 @@ func TestAccContainerNodePool_withCgroupMode(t *testing.T) {
...
+				Config: testAccContainerNodePool_withHugepageConfig(cluster, np, networkName, subnetworkName, 1),
+			},
+			{
+				ResourceName:      "google_container_node_pool.np",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Perform an update.
+			{
+				Config: testAccContainerNodePool_withHugepageConfig(cluster, np, networkName, subnetworkName, 1),
```
[the remark this answers is not in the document]
Comment (slevenick) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb:682]: Yeah, this should be addressed

**Review by slevenick — CHANGES_REQUESTED** [review]
The update test doesn't seem to do anything. Please fix so that the config changes

**Comment by DrFaust92** [comment]
thanks [slevenick](https://github.com/slevenick), added + covered second param in that block as well

**Review by slevenick — CHANGES_REQUESTED** [review]
Failing with:
        Error: Invalid expression
        
          on terraform_plugin_test.tf line 26, in resource "google_container_node_pool" "np":
          26: 	    hugepage_size_1g = %!d(MISSING)

**Comment by DrFaust92** [comment]
thanks [slevenick](https://github.com/slevenick), fixed + tested properly

**Pull request merged**